src/sdk/namespace_fs.js

```diff
@@ -811,6 +811,10 @@ class NamespaceFS {
                     }
                     for (let i = marker_index; i < sorted_entries.length; ++i) {
                         const ent = sorted_entries[i];
+                        if (list_versions && marker_curr) {
+                            const ent_name = _get_filename(ent.name);
+                            if (ent_name !== marker_curr) break;
+                        }
```

Comment:
I guess that after rebase this part would be deleted, right? (it is added in PR #8413 ).